nuto/mechanics/elements/ElementCollection.h

```diff
@@ -28,14 +28,17 @@ class ElementCollection
 //! semantics. Additionally, the compiler is free to eliminate all the copies (whenever that is the right thing to do).
 //! The access to the underlying elements is provided via const-reference. This reference is implicitly casted to the
 //! base class ElementInterface.
-template <typename TElement>
+template <typename TCoordinateElement, typename TDofElement>
 class ElementCollectionImpl : public ElementCollection
 {
 public:
-    static_assert(std::is_base_of<ElementInterface, TElement>::value,
-                  "TElement must be a descendant of ElementInterface");
+    static_assert(std::is_base_of<DofElementInterface, TDofElement>::value,
+                  "TElement must be a descendant of DofElementInterface");
 
-    ElementCollectionImpl(TElement coordinateElement)
+    static_assert(std::is_base_of<DofElementInterface, TDofElement>::value,
+                  "TElement must be a descendant of CoordinateElementInterface");
```

Comment:
This should probably say
~~~cpp
static_assert(std::is_base_of<CoordinateElementInterface, TCoordinateElement>::value,
                  "TCoordinateElement must be a descendant of CoordinateElementInterface");
~~~